pkgs/misc/vim-plugins/update.py

```diff
@@ -488,25 +470,49 @@ def parse_args():
         action="store_true",
         help="Automatically commit updates",
     )
+    parser.add_argument(
+        "--allow-dirty",
+        dest="allow_dirty",
+        action="store_true",
+        help=(
+            "Allow commit to continue even if state is unexpectedly dirty. "
+            "This is only helpful when developing vimPlugins infrastructure."
+        ),
+    )
 
     return parser.parse_args()
 
 
-def get_nixpkgs_repo() -> git.Repo:
-    repo = git.Repo(NIXPKGS_PATH)
-    if repo.is_dirty():
-        raise Exception("Please stash your changes before updating.")
-    return repo
+class NixpkgsRepo:
+    def __init__(self, allow_dirty: bool):
+        self.allow_dirty: bool = allow_dirty
+        self.repo: git.Repo = git.Repo(NIXPKGS_PATH)
 
+        if self.is_unexpectedly_dirty():
+            raise Exception("Please stash changes before updating.")
 
-def main() -> None:
-    args = parse_args()
-    if args.commit:
-        nixpkgs_repo = get_nixpkgs_repo()
-    plugin_names = load_plugin_spec(args.input_file)
-    current_plugins = get_current_plugins()
-    cache = Cache(current_plugins)
+    def commit(self, message: str, files: List[Path]) -> None:
+        file_paths = [str(f.resolve()) for f in files]
+        files_changed = False
+        for f in self.repo.index.diff(None):
+            if str(f) in file_paths:
+                files_changed = True
+                break
+
+        if files_changed:
+            print(f'committing to nixpkgs "vimPlugins: {message}"')
+            self.repo.index.add(file_paths)
+            self.repo.index.commit(f"vimPlugins: {message}")
+            assert self.is_unexpectedly_dirty() is False
+        else:
+            print("no changes in working tree to commit")
```

Comment:
More "pythonic" would probably be to just commit anyway and deal with the exception. It would definitely be easier.

Reply:
it isn't an exception, it would just create a useless commit with no diff.

Reply:
That's surprising, since the git CLI would complain about it.